### ISSUE TITLE
Can't build Strapi V5 Missing "./format/index.js" specifier in "date-fns" package

### ISSUE THREAD

**Issue by jdnielss** [strapi/strapi#22029]
### Environment
- Node.js version: v18.19.1
- NPM version: 10.2.4
- Strapi version: 5.1.0
- Database: Postgresql
- Operating system: Ubuntu 24
- Typescript 

### Strapi Report
Launched In: 7658 ms
Environment: development
OS: linux-x64
Strapi Version: 5.1.0
Node/Yarn Version: npm/10.2.4 node/v18.19.1 linux x64 workspaces/false
Edition: Community
Database: postgres

### Describe the bug

after npm run build, i got this error

```
✔ Compiling TS (13854ms)
⠋ Building build context
⠙ Building build context[INFO] Including the following ENV variables as part of the JS bundle:
    - ADMIN_PATH
    - STRAPI_ADMIN_BACKEND_URL
    - STRAPI_TELEMETRY_DISABLED
✔ Building build context (1000ms)
⠋ Building admin panel
✖ Building admin panel
[ERROR]  There seems to be an unexpected error, try again with --debug for more information 
│   [commonjs--resolver] Missing "./format/index.js" specifier in "date-fns" package       
````

### Steps to reproduce the behavior

1. npm install
2. npm run build

### Package.json

```
{
	"name": "strapi-be",
	"version": "0.1.0",
	"private": true,
	"description": "A Strapi application",
	"scripts": {
		"build": "strapi build",
		"deploy": "strapi deploy",
		"develop": "strapi develop",
		"start": "strapi start",
		"strapi": "strapi"
	},
	"dependencies": {
		"@_sh/strapi-plugin-ckeditor": "^4.0.6",
		"@strapi/plugin-cloud": "5.1.0",
		"@strapi/plugin-seo": "^2.0.1-rc.1",
		"@strapi/plugin-users-permissions": "5.1.0",
		"@strapi/strapi": "5.1.0",
		"pg": "8.8.0",
		"react": "^18.0.0",
		"react-dom": "^18.0.0",
		"react-router-dom": "^6.0.0",
		"styled-components": "^6.0.0"
	},
	"devDependencies": {
		"@types/node": "^20",
		"@types/react": "^18",
		"@types/react-dom": "^18",
		"typescript": "^5"
	},
	"engines": {
		"node": ">=18.0.0 <=22.x.x",
		"npm": ">=6.0.0"
	},
	"strapi": {
		"uuid": "c154695c-6b01-462a-b495-1e2efe62d599"
	}
}

```

**Comment by mkmule** [comment]
It looks to be related to the fact that: `  "name": "@strapi/admin",` requires `"date-fns": "2.30.0",`, while `  "name": "@strapi/plugin-seo",` requires `    "date-fns": "^4.1.0",`, hence version mismatch. 

I am getting the same issue when doing docker build, for now, I managed to solve it using node v20 in my Dockerfile:
```
# Creating multi-stage build for production
FROM node:20-alpine
...
```

TLDR: Either stapi/admin (maybe other packages) are updated to use v4 of date-fns, either seo plugin downgrades to the same v2 version as strapi/admin.
